app/controllers/application_controller.rb

```diff
@@ -1,5 +1,16 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  before_action :authenticate_user!
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:name, :photo, :bio, :email, :password, :post_counter)
```

Comment:
- [ ] Post counter is not required information for signing up. Kindly remove it.

- [ ] I am unable to sign up. You did not provide the input box for `name`, yet you are validating for it. Kindly fix that.

  Since I am unable to sign up in the first place I am unable to check other requirements for this project so ensure other listed requirements are met and that your app works correctly

  ![image](https://user-images.githubusercontent.com/36542986/225284862-cfdbf9f0-8012-4891-bc31-eb2f011ad2cf.png)

Reply:
thanks. it's done!